frb_codegen/src/utils.rs

```diff
@@ -12,15 +12,69 @@ pub fn mod_from_rust_path(code_path: &str, crate_path: &str) -> String {
         .replace('/', "::")
 }
 
-pub fn with_changed_file<F: FnOnce() -> anyhow::Result<()>>(
+pub fn with_changed_file(
     path: &str,
     append_content: &str,
-    f: F,
+    generate_c_from_rust_func: impl FnOnce() -> anyhow::Result<()>,
+    generate_dart_from_c_func: impl FnOnce() -> anyhow::Result<()>,
 ) -> anyhow::Result<()> {
     let content_original = fs::read_to_string(&path)?;
     fs::write(&path, content_original.clone() + append_content)?;
 
-    f()?;
+    generate_c_from_rust_func()?;
+    generate_dart_from_c_func()?;
 
     Ok(fs::write(&path, content_original)?)
 }
+
+pub fn find_all_duplicates(mut vec: Vec<String>) -> anyhow::Result<()> {
```

Comment:
I guess no need for so long code. As for duplicates, what about utilizing `Set`s? https://stackoverflow.com/questions/46766560/how-to-check-if-there-are-duplicates-in-a-slice

Sth like:

```
let mut uniq = HashSet::new();
let mut duplicates = vec![];
iter.into_iter().all(move |x| {
 let already_exists = uniq.insert(x);
 if already_exists { duplicates.insert(x) }
})
if duplicates.isnotempty() { panic (...) }

```

Btw, try to make this function generic as it is so general

Reply:
fixed and make it generic in a96cb32930e83bfddb9db35f896f058f8a7d2e36.